hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.monster.dap
+
+import org.broadinstitute.monster.dogaging.jadeschema.table.HlesHealthCondition
+
+object HealthTransformations {
+
+  /** Parse all health-condition-related fields out of a raw RedCap record. */
+  def mapHealthConditions(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    // condition type is congenital vs infectious vs tox vs trauma vs etc
+    health_conditions.filter { case (condition, _) => rawRecord.getBoolean(s"hs_${condition}_yn") }.flatMap {
+      // infectious disease
+      case ("dx_infectious", conditionCategorical) =>
+        // process infectious disease data
+        infectious_diseases.filter {
+          case (infection, _) => rawRecord.getBoolean(s"hs_dx_${infection}")
+        }.map {
+          case (infection, infectionCategorical) =>
+            HlesHealthCondition(
+              dogId = rawRecord.getRequired("study_id").toLong,
+              hsConditionType = conditionCategorical.toLong,
+              hsCondition = infectionCategorical.toLong,
+              // should only be populated if hs_dx_infect_other is the current infectionCategorical (value is 40)
+              hsConditionOtherDescription = if (infectionCategorical == 40) {
+                rawRecord.getOptional("hs_dx_infect_other_spec")
+              } else { None },
+              // infectious diseases are not categorized as congenital
+              hsConditionIsCongenital = false,
+              // no relevant field for infectious diseases
+              hsConditionCause = None,
+              // no relevant field for infectious diseases
+              hsConditionCauseOtherDescription = None,
+              hsDiagnosisYear = rawRecord.getOptionalNumber(s"hs_dx_${infection}_year"),
+              hsDiagnosisMonth = rawRecord.getOptionalNumber(s"hs_dx_${infection}_month"),
+              hsRequiredSurgeryOrHospitalization =
+                rawRecord.getOptionalNumber(s"hs_dx_${infection}_surg"),
+              hsFollowUpOngoing = rawRecord.getOptionalBoolean(s"hs_dx_${infection}_fu")
+            )
+        }
+      // an "else" case to please the compiler gods for now, will eventually be replaced by all the other health
+      // condition cases
+      case (_, conditionCategorical) =>
```

Comment:
I don't think this pattern is scalable (there will be much repeated code across cases, and the file will be looooong). IMO it'd be better to have:
1. A generic, parameterized method that handles the actual `RawRecord` => `Option[HlesHealthCondition]` conversion for a specific condition
2. A method per each condition_type which handles the top-level Y/N filter, looping over each specific condition (delegating to the generic method), and adding any edge-cases (i.e. the "other" spec case)